engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -62,17 +62,28 @@ func CreateEnclavePool(
 		)
 	}
 
-	// The amount of idle enclaves is equal to the chan capacity + one enclave (this one will be waiting in the queue until the channel is unblocked)
-	// chanCapacity = 0  means that it will be an unbuffered, so the communication between the channels will succeed when both (sender and receiver) are ready
-	chanCapacity := poolSize - oneEnclave
+	// this channel is the repository of idle enclave UUIDs
+	idleEnclavesChan := make(chan enclave.EnclaveUUID, poolSize)
+
+	// this channel is for communicate to the sub-routine that one
+	// idle enclave has been picked from the pool, so it can trigger the
+	// fill mechanism again, it has the capacity = poolSize for not blocking
+	// the caller
+	fillChan := make(chan bool, poolSize)
```

Comment:
yes here it makes sense to use a channel. However i don't understand why it needs to be buffered. If it's indeed used a signal to send to sub-routine that it needs to start a new idle enclave. I don't think this should ever get more than one value, unless i'm missing something

Reply:
it's buffered for handling concurrent requests and not blocking any of them.